src/Microsoft.ML.TimeSeries/RootCauseAnalyzer.cs

```diff
@@ -702,7 +702,7 @@ private static bool ContainsAll(Dictionary<string, Object> bigDictionary, Dictio
 
         private bool IsAggregationDimension(Object val, Object aggSymbol)
         {
-            return Convert.ToString(val).Equals(aggSymbol);
+            return Object.Equals(val, aggSymbol);
```

Comment:
>Object.Equals(val, aggSymbol); [](start = 19, length = 30)

please add unti test to verify the fix, example: comparing between: 1 to "1", 1 to 1, "1" to "1" to verify we can correctly verify same type and value

Reply:
add more test cases